src/derivation.ts

```diff
@@ -41,38 +52,42 @@ type DeriveKeyFromPathArgs = {
  * WARNING: It is the consumer's responsibility to ensure that the path is valid
  * relative to its parent key.
  *
- * @param path - A full or partial HD path, e.g.:
+ * @param args
+ * @param args.path - A full or partial HD path, e.g.:
  * bip39:SEED_PHRASE/bip32:44'/bip32:60'/bip32:0'/bip32:0/bip32:0
  *
  * BIP-39 seed phrases must be lowercase, space-delimited, and 12-24 words long.
- * @param privateKey - The parent key of the given path segment, if any.
- * @param publicKey - The parent public key of the given path segment, if any.
- * @param chainCode - The chain code of the given path segment, if any.
- * @param depth - The depth of the segment.
- * @param curve - The curve to use.
+ * @param args.node - The node to derive from.
+ * @param args.depth - The depth of the segment.
  * @returns The derived key.
  */
-export async function deriveKeyFromPath({
-  path,
-  privateKey,
-  publicKey,
-  chainCode,
-  depth,
-  curve,
-}: DeriveKeyFromPathArgs): Promise<DerivedKeys> {
-  if (privateKey && !Buffer.isBuffer(privateKey)) {
-    throw new Error('Private key must be a Buffer if specified.');
+export async function deriveKeyFromPath(
+  args: DeriveKeyFromPathArgs,
+): Promise<SLIP10Node> {
+  const { path, depth = path.length } = args;
+
+  const node = 'node' in args ? args.node : undefined;
+  const curve = 'curve' in args ? args.curve : node?.curve;
+
+  if (!curve) {
```

Comment:
Do we not need to check the validity of `node` somewhere here? We had checks for the private key validity before.